### ISSUE TITLE
Unknown DW_FORM_data16

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#12744]
I'm seeing some very weird behaviour with static builds
This is coming out of STDERR

```
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (128)
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (128)
Failed to raise an exception: END_OF_STACK
[0x458246] ???
[0x42ce58] ???
[0x4395ae] ???
[0x433ec9] ???
[0xcf6348] ???

Tried to raise:: Unknown DW_FORM_data16 (Exception)
  from usr/lib/crystal/core/crystal/dwarf/info.cr:83:29 in '??'
  from usr/lib/crystal/core/crystal/dwarf/info.cr:67:23 in '??'
  from usr/lib/crystal/core/exception/call_stack/elf.cr:10:7 in '??'
  from src/ldso/dl_iterate_phdr.c:45:1 in '??'
```

That's running an app and making request built using a this Dockerfile:
https://github.com/spider-gazelle/spider-gazelle/blob/7c360488ad7b5aef93b92789cd08b8a2ddf03265/Dockerfile

the Unsupported BIO ctrl call reminds me of this issue: #12025

The app runs and handles requests, eventually failing with `END_OF_STACK` and exiting

_Originally posted by @stakach in https://github.com/crystal-lang/crystal/issues/6288#issuecomment-1240167192_

**Comment by straight-shoota** [comment]
I found the warnings occur only when using the edge version of OpenSSL - so that's resolved.
So I assume there are few more cases that need to be handled in the near future

```
WARNING: Unsupported BIO ctrl call (133)
WARNING: Unsupported BIO ctrl call (128)
```

Still seeing the `Tried to raise:: Unknown DW_FORM_data16 (Exception)` on static builds

*Originally posted by @stakach in https://github.com/crystal-lang/crystal/issues/6288#issuecomment-1240183744*

**Comment by straight-shoota** [comment]
Can confirm, I also see `Tried to raise:: Unknown DW_FORM_data16 (Exception)`
on static builds with Crystal 1.6.2.

Unfortunately don't have time to extract an isolated test-case atm
but the trigger is definitely when an Exception gets raised inside a Fiber.

I can trigger it by sending malformed data to a TCPServer (not http/server!),
so that an Exception gets raised in the handler-fiber.

Instead of entering the `rescue`-block it then crashes with above message.
(possibly while trying to pass the Exception upwards across Fiber-boundaries,
but can't say that for sure - haven't dug deeper into it, yet).

It doesn't happen when compiling without `--static`, so that's my workaround for now.

*Originally posted by @m-o-e in https://github.com/crystal-lang/crystal/issues/6288#issuecomment-1311130315*

**Comment by stakach** [comment]
I'm not seeing the error on the latest builds anymore - pretty sure it's resolved

**Comment by straight-shoota** [comment]
Ah right that was fixed in #12497

But @m-o-e reported this for 1.6.2 as well in https://github.com/crystal-lang/crystal/issues/6288#issuecomment-1311130315

**Comment by m-o-e** [comment]
Ugh. Apologies!

I just realized I was still on 1.5.1 with that build, not on 1.6.2.
Just retried with 1.6.2 and can no longer reproduce it.

So please ignore / redact my confirmation above, false alert. 🙈
